sklearn/preprocessing/label.py

```diff
@@ -352,8 +352,7 @@ def fit_transform(self, y):
         Y : numpy array or CSR matrix of shape [n_samples, n_classes]
             Shape will be [n_samples, 1] for binary problems.
         """
-        self.fit(y)
-        return self.transform(y)
+        return super(fit_transform, self).fit_transform(y)
```

Comment:
I'm not sure, but I think flake8 may fail if compile fails. I haven't used it very much. I get a connection error when I try to access the travis-ci report, so I can't look at it. I'll go ahead and fix my bug.